lib/PreviewProvider.js

```diff
@@ -342,18 +298,16 @@ PreviewProvider.prototype = {
         let responseJson = yield response.json();
         this._tabTracker.handlePerformanceEvent(event, "embedlyProxyRequestReceivedCount", responseJson.urls.length);
         this._tabTracker.handlePerformanceEvent(event, "embedlyProxyRequestSucess", 1);
-        let currentTime = Date.now();
-        newLinks.forEach(link => {
+
+        for (let link of newLinks) {
           let data = responseJson.urls[link.sanitized_url];
           if (!data) {
             return;
           }
-          ss.storage.embedlyData[link.cache_key] = Object.assign({}, ss.storage.embedlyData[link.cache_key], data);
-          if (updateAccessTime) {
-            ss.storage.embedlyData[link.cache_key].accessTime = currentTime;
-          }
-          ss.storage.embedlyData[link.cache_key].refreshTime = currentTime;
-        });
+          // Save the entry in the metadata database
+          let entry = Object.assign({}, link, {expired_at: (this.options.metadataTTL) + Date.now()});
+          yield this._metadataStore.asyncInsert([Object.assign({}, data, entry)]);
```

Comment:
perhaps @ncloudioj could clarify if we need to yield - i think it has to do with making sure we insert uniques (race conditions if we don't make sure we've inserted that link before requesting to insert that same one)